modin/data_management/factories/dispatcher.py

```diff
@@ -14,7 +14,7 @@
 """
 Contain IO dispatcher class.
 
-Dispatcher routes the work to excecution engine-specific functions.
+Dispatcher routes the work to backend-specific functions.
```

Comment:
Dispatcher dispatch calls to the factories, each factory represents some *backend*, so it's routing work rather to backends than the engines.